### DISCUSSION TITLE
Nuxt 3 active link on parent

### DISCUSSION THREAD

**Discussion by wadethomastt** [nuxt/nuxt#16003] [Questions] [answered]
HI,

I have successfully implemented the active class on my nav links for the current page. The issue I'm having is that if I go to a child page the parent is no longer active. I'm not sure if that is how its suppose to behave. My code is simple but I just can understand what I'm missing.

CSS File

@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;700;900&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Playfair+Display:wght@400;700;900&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Play:wght@400;700&display=swap");

@tailwind base;
@tailwind components;
@tailwind utilities;

/* home route and active route will show in bold as it matches / and /about */
.router-link-active {
  font-weight: bold;
}

Nav File
<template>
  <div class="container mx-auto">
    <nav class="flex justify-center items-center pt-4">
      <div class="flex space-x-16 text-sm text-stone-700">
        <NuxtLink to="/about" class="">About</NuxtLink>
        <NuxtLink to="/bible">Bible</NuxtLink>
        <NuxtLink to="/resources">Resources</NuxtLink>
        <NuxtLink to="/donate">Donate</NuxtLink>
        <NuxtLink to="/contact">Contact</NuxtLink>
      </div>
    </nav>
  </div>
</template>

**Comment by TheAlexLichter** [comment]
Hey! This should work fine - see https://github.com/vuejs/rfcs/blob/master/active-rfcs/0028-router-active-link.md#nested-routes

Please provide a minimal reproduction (e.g. via https://nuxt.new) that highlights the issue.

**Reply by wadethomastt** [reply]
Ok thanks for the link I'm checking it out. Don't fully understand everything but will see if it helps thanks.

**Comment by k16e** [comment]
Hmm, I just now ran into a similar issue and I'm kinda bummed reading through the doc, looking like Vue Router no longer keeps track of "children" routes. I always love to indicate the navigation item as "active" when a "child" route has to plain "to" string of the nav item. For example, I love to keep the link === `/products/` active when any `/products/product-[id]` is the current page. Now I don't see that VR or N3 automatically adds the active class... is there a workaround? It was/is a needed feature.

**Reply by AndreasHerss** [reply]
For what it's worth, i found that the page structure is what makes it break.
So if i use the structure like so:
account/
-index.vue
-[id].vue

Then /account is not active when accessing the /account/[id] route

But if i use the following structure, it works

account.vue
account/
-[id].vue

So making the index.vue file to be named the same as the folder works. 
The reason i see is that vue-router (or nuxt) doesn't nest the routes as expected when using a index.vue file in the folder

**Comment by lihbr** [comment, marked as answer]
Hey there, I'm gonna quote my answer [from another thread](https://github.com/nuxt/framework/discussions/2086#discussioncomment-4575574) that chimes with what @manniL already mentioned, have a look at the workaround section for how to easily implement back such a behavior on header links :)

---

## Why is it behaving like this?

After some research to confirm, it appears this is not a Nuxt bug, but Vue Router's intended behavior. Vue Router 4 came with a change to how `router-link-active` is applied.

> Routes that are unrelated from a record point of view but share a common path are no longer active.
>
> [View full `router-link-active` RFC](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0028-router-active-link.md)

In Vue Router's language, "unrelated" means that routes don't share a parent/children relationship in the route definition array.

```typescript
// Unrelated, no parent/children relationship
const routes = [
	{ path: "/foo" },
	{ path: "/foo/bar" },
];

// Related, has a parent/children relationship
const routes = [
	{
		path: "/foo",
		children: [
			{ path: "/bar" },	// inferred as `/foo/bar`
		],
	},
];
```

The parent/children relationship on routes is meant for a specific use case allowing component nesting based on the URL ([see Vue Router's Nested Routes documentation](https://router.vuejs.org/guide/essentials/nested-routes.html), which is also [available with Nuxt](https://nuxt.com/docs/guide/directory-structure/pages#nested-routes)), but I don't think that's what you want in this case (at least that's not what I'd naturally lean towards)

## Workarounds

Bruno's workaround works, similarly Eduardo and Vue Router's team shared a workaround in the [RFC mentioned earlier](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0028-router-active-link.md#motivation).

```html
<router-link v-slot="{ route }">
  partial path: {{ $route.path.startsWith(route.path) }}
  <br />
  partial path + partial query: {{ $route.path.startsWith(route.path) && includesQuery($route.query, route.query) }}
</router-link>
```

Which in `<NuxtLink/>`'s case could be applied like so.

```html
<nuxt-link to="/blog" :class="{ 'parent-active-class': $route.path.startsWith('/blog') }">
	...
</nuxt-link>
```

## The future

Vue Router's `router-active-link` RFC mentioned that to make the old active class behavior replicable.

> We could maybe allow `component` to be absent [from a route record] and internally behave as if there where a `component` option that renders a `RouterView` component.

[This has been implemented](https://github.com/vuejs/router/issues/1174), however, with Nuxt's file system-based routing it would have been hard to take advantage of it anyway.

We *could* bake this behavior within `<NuxtLink />` layer over `<RouterLink />` (either as a flag to enable it, or as a 3rd kind of active class), however, I'm not really sure about it as I think it might be too specific(?) and the motivations listed in the RFC are strong to me.
